datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -959,14 +953,18 @@ pub async fn fetch_parquet_metadata(
     store: &dyn ObjectStore,
     meta: &ObjectMeta,
     size_hint: Option<usize>,
-    decryption_properties: Option<&FileDecryptionProperties>,
+    #[allow(unused)] decryption_properties: Option<&FileDecryptionProperties>,
 ) -> Result<ParquetMetaData> {
     let file_size = meta.size;
     let fetch = ObjectStoreFetch::new(store, meta);
 
-    ParquetMetaDataReader::new()
-        .with_prefetch_hint(size_hint)
-        .with_decryption_properties(decryption_properties)
+    #[allow(unused_mut)]
+    let mut reader = ParquetMetaDataReader::new().with_prefetch_hint(size_hint);
```

Comment:
I don't think this mut is needed even with `parquet_encryption` enabled, as `with_decryption_properties` takes ownership of the reader.

```suggestion
    let reader = ParquetMetaDataReader::new().with_prefetch_hint(size_hint);
```

Reply:
Good catch. Updated.